apache/superset
# feat(explore): Move chart header to top of the page

### SUMMARY
This PR implements part of chart header redesign (final design available here: https://github.com/apache/superset/discussions/19099)

List of changes:
- move chart header to top of the page
- change looks and behaviour of editable title - input grows when user types text, no wrapping, when title overflows it gets truncated and tooltip with full name appears on hover
- when title is truncated, clicking (enabling editing mode) moves the cursor to the end of the title
- Applied additional paddings to Dataset label and column search on the left so that everything is aligned
- Deleting the whole title is allowed - we don't revert to the last title when user clears the input. Instead, a placeholder is shown
- When new chart is created, the default title is empty instead of "- untitled", which means that we display the placeholder
- Editing the name of existing chart gets recorded in "Altered" table
- if user doesn't have edit permissions, instead of an input element display span element that looks and behaves the same except for editing capabilities

Next steps:
- move timer, rows retrieved and cached labels to chart container (implemented in https://github.com/apache/superset/pull/19458)
- move save button to the right of chart header 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/15073128/161726241-ffdf4019-4904-4ef0-8efd-b0fa749fdd36.mov


### TESTING INSTRUCTIONS
1. Open a chart that you have permissions to edit
2. Click the title, verify that cursor appears
3. When there's no text a placeholder should appear
4. When you start typing, the input's width should adjust to its content
5. When input is wider than the header container, it should move to the position of cursor if in edit mode, or be truncated if not in edit mode
6. If not editing and title is truncated, display tooltip with full title on hover 


### ADDITIONAL INFORMATION


CC @kasiazjc
